userprefs.php

```diff
@@ -924,7 +924,14 @@ function _show_textfield( $field_name, $current_value, $extras )
 {
     list($size, $rest) = $extras;
     $current_value_esc = attr_safe($current_value);
-    echo "<input type='text' name='$field_name' value='$current_value_esc' size='$size'>$rest";
+    echo "<input type='text' style='width: ${size}em' name='$field_name' value='$current_value_esc'>$rest";
+}
+
+function _show_numberfield( $field_name, $current_value, $extras )
```

Comment:
https://github.com/DistributedProofreaders/dproofreaders/blob/a4f9aa136899dfd0b9ce52519e203be257c1c326/userprefs.php#L552-L567

These two could be `numberfield` too.

Might be a good idea to add `required` to all of them since it throws `Fatal error: Uncaught InvalidArgumentException` if one of them was not given. 

Might be a good idea to add `min=1` to most (if not all of them) since negative values make no sense in this context. 

Handles `5.0` and `5e0` well.

Reply:
Good call on those two fields that I missed. I've fixed those and added `min=1` to all of them.